ospfd/ospf_lsa.c

```diff
@@ -3648,3 +3648,19 @@ int ospf_lsa_refresh_walker(struct thread *t)
 
 	return 0;
 }
+
+void ospf_nssa_flush_external_lsa(struct ospf *ospf, struct in_addr area_id)
+{
+	struct ospf_area *area;
+	struct route_node *rn;
+	struct ospf_lsa *lsa;
+
+	area = ospf_area_get(ospf, area_id);
+
+	if (area->external_routing == OSPF_AREA_NSSA) {
+		LSDB_LOOP (EXTERNAL_LSDB(ospf), rn, lsa)
+			if (IS_LSA_SELF(lsa) &&
+			    !(CHECK_FLAG(lsa->flags, OSPF_LSA_LOCAL_XLT)))
+				ospf_lsa_flush_area(lsa, area);
```

Comment:
External LSAs have domain flooding scope, so why not extending the flush to the domain instead of area only ? otherwise the rest of routers in the domain will not be updated

Reply:
The maxage LSA is to be sent on all interfaces attached to the area which is configured as NSSA.  Therefore can be other interfaces in default area , the maxage LSA should not be sent on these interfaces

Reply:
I think this would cause a domain-side flood anyway?

Reply:
ospf_lsa_flush_area() will flood maxage LSA on all the interfaces attached to the area (specified in second parameter)
This will ensure all neighbors in that area will get maxage LSA and flush their databases